File: src/test/python/NestedTypesAction.py
from TrifaceJythonAction import TrifaceJythonAction

# Simple class to test java -> jython integration
# Sets basic python types
class NestedTypesAction(TrifaceJythonAction):

  def execute(self):
    self.objectmap['list'] = [["a","b"],["c","d"]]
    self.objectmap['map'] = {'key1': {'foo' : 'bar'}, 'key2': {'bar' : 'baz'}}
    return self.getMap()

File: src/test/js/NestedTypesAction.js
// Simple class to test java -> JS integration
// Sets basic JS types
action.execute=function(){
  
  this.objectmap.put("list", [["a","b"],["c","d"]]);
  
  m1 = new HashMap();
  m1.put("key1","foo");
  m1.put("key2","bar");
  
  m2 = new HashMap();
  m2.put("key1", "bar");
  m2.put("key2", "baz");
  
  outerMap = new HashMap();
  outerMap.put("key1",m1);
  outerMap.put("key2",m2);
  
  this.objectmap.put("map", outerMap);
  
  return this.getMapInternal();
};
